assignments/prototypes.js

```diff
@@ -102,9 +137,92 @@
   console.log(archer.greet()); // Lilith offers a greeting in Elvish.
   console.log(mage.takeDamage()); // Bruce took damage.
   console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
 
   // Stretch task: 
   // * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.  
   // * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villain and one a hero and fight it out with methods!
\ No newline at end of file
+  // * Create two new objects, one a villain and one a hero and fight it out with methods!
+
+  function HeroVillan(attributes) {
+    Humanoid.call(this, attributes);
+      this.ability = attributes.ability;
+      this.finish = attributes.finish;
+  }
+  
+  HeroVillan.prototype = Object.create(Humanoid.prototype);
+  
+  HeroVillan.prototype.fight1 = function () {
+    return `${this.name} lashes out with the ${this.ability} ability.`;
+  };
+
+  HeroVillan.prototype.fight2 = function (target) {
+    if (target.healthPoints <= 0){
+    return `${this.name} finishes ` + target.name + ` with ${this.finish}!!!`;
+    }
+    else {
+    return `${this.name} swings his ${this.weapons} at ` + target.name;
+  }
+  };
+
+  const hero = new HeroVillan({
+    createdAt: new Date(),
+    dimensions: {
+      length: 2,
+      width: 2,
+      height: 3,
+    },
+    healthPoints: 50,
+    name: 'Jin Kazama',
+    team: 'Mishima Zaibatsu',
+    weapons: [
+      'Chain',
+    ],
+    language: [
+      'Common',
+      'Demonic',
+      'Japanese',],
+    ability: 'Lightning Screw Uppercut',
+    finish: 'Mishima Style 10-Hit Combo',
+  });
+
+  const villan = new HeroVillan({
+    createdAt: new Date(),
+    dimensions: {
+      length: 2,
+      width: 2,
+      height: 4,
+    },
+    healthPoints: 50,
+    name: 'Kazuya Mishima',
+    team: 'Mishima Zaibatsu',
+    weapons: [
+      'Fists',
+    ],
+    language: [
+      'Common',
+      'Demonic',
+      'Japanese',],
+    ability: 'Spinning Demon',
+    finishingmove: 'Demonic Laser Beam',
+  });
+
+  // console.log(villan.fight1());
+  // console.log(hero.fight1());
+  // console.log(hero.fight2(villan));
+  // console.log(villan.fight2(hero));
+
+  // villan["healthPoints"] = -1;
+
+  // console.log(villan.healthPoints);
+  // console.log(villan);
+  
+  // console.log(hero.fight2(villan))
+
+  console.log('Round 1 Fight!')
+  console.log(villan.fight1());
+  console.log(hero.fight1());
+  console.log(hero.fight2(villan));
+  console.log(villan.fight2(hero));
+  console.log(villan.name + ' is looking hurt. Finish Him!')
+  villan["healthPoints"] = -1;
```

Comment:
you're manually adjusting the villan's health points to -1.  As an added stretch.  Can you adjust his health within a prototypal function?

Reply:
You're going to laugh but, I literally fixed that when I woke up this evening. Not even Kidding... It literally came to me in one of those -early morning trips to the bathroom-